src/main/java/org/generationcp/middleware/dao/AttributeDAO.java

```diff
@@ -119,70 +87,69 @@
 		return returnList;
 	}
 
-	@SuppressWarnings("unchecked")
-	public List<UserDefinedField> getAttributeTypes() {
-		final Criteria criteria = this.getSession().createCriteria(UserDefinedField.class).add(Restrictions.eq("ftable", "ATRIBUTS"));
-		return criteria.list();
-	}
-
-	public Attribute getAttribute(final Integer gid, final String attributeName) {
+	public Attribute getAttributeByGidAndVariableId(final Integer gid, final Integer variableId) {
 		Attribute attribute = null;
 		try {
-			final String sql = "SELECT {a.*} FROM atributs a INNER JOIN udflds u ON (a.atype=u.fldno)"
-				+ " WHERE a.gid = :gid AND u.ftable='ATRIBUTS' and u.fcode=:name";
+			final String sql = "SELECT {a.*} FROM atributs a "
+				+ " WHERE a.gid = :gid AND a.atype = :variableId";
 			final SQLQuery query = this.getSession().createSQLQuery(sql);
 			query.addEntity("a", Attribute.class);
 			query.setParameter("gid", gid);
-			query.setParameter("name", attributeName);
+			query.setParameter("variableId", variableId);
 			final List<Attribute> attributes = query.list();
 			if (!attributes.isEmpty()) {
 				attribute = attributes.get(0);
 			}
 
 		} catch (final HibernateException e) {
-			throw new MiddlewareQueryException("Error with getAttribute(gidList=" + gid + ", " + attributeName + "): " + e.getMessage(), e);
+			throw new MiddlewareQueryException(
+				"Error with getAttributeByGidAndVariableId(gid=" + gid + ", " + variableId + "): " + e.getMessage(), e);
 		}
 		return attribute;
 	}
 
-	public List<GermplasmAttributeDto> getGermplasmAttributeDtos(final Integer gid, final String attributeType) {
+	public List<GermplasmAttributeDto> getGermplasmAttributeDtos(final Integer gid, final Integer variableTypeId, final String programUUID) {
 		try {
 			final StringBuilder queryString = new StringBuilder();
 			queryString.append("Select a.aid AS id, ");
+			queryString.append("cv.cvterm_id as variableId, ");
 			queryString.append("a.aval AS value, ");
-			queryString.append("u.fcode AS attributeCode, ");
-			queryString.append("u.ftype AS attributeType, ");
-			queryString.append("u.fname AS attributeDescription, ");
+			queryString.append("IFNULL(vpo.alias, cv.name) AS variableName, ");
+			queryString.append("cp.value AS variableTypeName, ");
+			queryString.append("cv.definition AS variableDescription, ");
 			queryString.append("CAST(a.adate AS CHAR(255)) AS date, ");
 			queryString.append("a.alocn AS locationId, ");
 			queryString.append("l.lname AS locationName ");
 			queryString.append("FROM atributs a ");
-			queryString.append("INNER JOIN udflds u ON a.atype = u.fldno ");
+			queryString.append("INNER JOIN cvterm cv ON a.atype = cv.cvterm_id ");
```

Comment:
Better to add in join to cvterm cv the clause: 
`and c.cv_id = "+ CvId.VARIABLES.getId()`